fabric-chaincode-shim/src/main/java/org/hyperledger/fabric/contract/execution/ExecutionFactory.java

```diff
@@ -36,10 +36,7 @@ public InvocationRequest createRequest(final ChaincodeStub context) {
      * @param serializers Instance of the serializer
      * @return Execution Service
      */
-    public ExecutionService createExecutionService(final SerializerInterface serializers) {
-        if (es == null) {
-            es = new ContractExecutionService(serializers);
-        }
-        return es;
+    public ExecutionService createExecutionService(final SerializerRegistryImpl serializers) {
+        return new ContractExecutionService(serializers);
```

Comment:
Thinking of performance and the cost of the `new ContractExecutionService` being created each transaction.
Like you've followed the registry pattern - now that there is the Serializer Register - and the contents of that won't change at runtime.. can the ContractExecutionService not be cached?

Reply:
With proposed changes this method should only be called when initializing a new instance of _ContractRouter_ (see changes in the constructor and the new field _executor_). So, a new _ExecutionService_ won't be created for each transaction, but there will be a single instance for each _ContractRouter_ that will be used for all requests.